aspnetcore/blazor/Security-guidance.md

```diff
@@ -0,0 +1,254 @@
+}
+```
+
+By adding the `count < 3` check inside the handler we force the decission to be taken based on the current app state, and not the state of the UI which might be out of date.
+
+# Other security recommendations
+In general, all the recommendations for securing ASP.NET Core applications apply to blazor server-side applications.
+
+In this section we will go through some of the common application threats, explain what blazor server-side does to address it and what recommended application specific actions can be taken to further protect your application.
+
+## Logging and sensitive data
+Blazor server-side applications interact heavily with the client through events and JS interop. All these interactions are recorded on the server logs via the use of `Microsoft.AspNetCore.Extensions.Logging.ILogger`. Blazor avoids logging sensitive information, like actual events or JS interop inputs/outputs.
+
+When an error occurs on the server we notify the client and tear down the session. By default the client receives a generic error message that can be read when opening the developer tools.
+
+This error does not include any callstack and doesn't provide any details of what the cause of the error was. The server logs contain such information and for development purposes it can be made available to the client by turning on detailed errors.
+
+This can be done either on `CircuitOptions.DetailedErrors` or by setting the `DetailedErrors` configuration key, for example through the `ASPNETCORE_DETAILEDERRORS=true` environment variable.
+
+## Protecting information in transit
+Blazor server-side uses SignalR under the hood to comunicate between the client and the server. That means that blazor server-side will normally use whatever transport SignalR negotiates (typically websockets).
+
+Blazor server-side doesn't take any extra step to ensure the integrity and confidentiality of the data sent between the server and the client.
+
+For that reason, the recommendation is to use HTTPS for communications if your app is not in a trusted environment.
+
+## Cross-site scripting (XSS)
+This threat allows an unauthorized party to execute arbitrary logic in the context of the browser. A compromised application would potentially run arbitrary code on the client (user's browser) and such vulnerability could be used to potentially perform a number of actions against the server. Such actions include:
+* Dispatching fake/invalid events to the server.
+* Dispatching fail/invalid render completions.
+* Avoid dispatching render completions.
+* Dispatching interop calls from JS to .NET.
+* Modifying the response of interop calls from .NET to JS.
+* Avoid dispatching .NET to JS interop results.
+
+Blazor server-side takes some steps towards protecting against some of these threats but ultimately your application needs to be aware of these and take appropriate actions:
+* Don't trust data on event handlers before validating it first.
```

Comment:
I'm not sure how this one relates to XSS. It seems to be a general "don't trust JS" sentiment, which although true, is covered elsewhere in this doc.